app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -336,34 +328,40 @@ public void handleTaskCompletion(ResultAndError<AppInstallStatus> result) {
         uiController.refreshView();
     }
 
-    private void reportFailureToNotifications(MessageTag message, NotificationActionButtonInfo.ButtonAction buttonAction) {
-        CommCareApplication.notificationManager()
-                .reportNotificationMessage(NotificationMessageFactory.message(message, buttonAction), true);
-        uiController.setNotificationsVisible();
-    }
-
-    private void reportFailureToNotifications(String errorMessage, NotificationActionButtonInfo.ButtonAction buttonAction) {
+    private void reportFailureToNotifications(AppInstallStatus status, String errorMessage) {
         NotificationMessage notificationMessage = null;
 
-        if (UpdateHelper.isCombinedErrorMessage(errorMessage)) {
-            Pair<String, String> resourceAndMessage =
-                    UpdateHelper.splitCombinedErrorMessage(errorMessage);
-            notificationMessage =
-                    NotificationMessageFactory.message(AppInstallStatus.InvalidResource,
-                            new String[]{null, resourceAndMessage.first, resourceAndMessage.second},
-                            buttonAction);
-        } else if (!"".equals(errorMessage)) {
-            notificationMessage =
-                    NotificationMessageFactory.message(AppInstallStatus.UpdateFailedGeneral,
-                            new String[]{null, errorMessage, null},
-                            buttonAction);
-        }
+        switch (status) {
+            case BadSslCertificate:
+                notificationMessage = NotificationMessageFactory.message(status,
+                        NotificationActionButtonInfo.ButtonAction.LAUNCH_DATE_SETTINGS);
+                break;
+            //Other special handlers will go here when implemented
+            default:
+                if (UpdateHelper.isCombinedErrorMessage(errorMessage)) {
+                    Pair<String, String> resourceAndMessage =
+                            UpdateHelper.splitCombinedErrorMessage(errorMessage);
+                    notificationMessage =
+                            NotificationMessageFactory.message(AppInstallStatus.InvalidResource,
+                                    new String[]{null, resourceAndMessage.first, resourceAndMessage.second},
+                                    NotificationActionButtonInfo.ButtonAction.NONE);
+                } else if (!"".equals(errorMessage)) {
+                    notificationMessage =
+                            NotificationMessageFactory.message(AppInstallStatus.UpdateFailedGeneral,
+                                    new String[]{null, errorMessage, null},
+                                    NotificationActionButtonInfo.ButtonAction.NONE);
+                } else {
+                    notificationMessage =
+                            NotificationMessageFactory.message(status,
+                                    NotificationActionButtonInfo.ButtonAction.NONE);
```

Comment:
Is this a change in behaviour from earlier i.e. we are showing a notification even when errorMessage is empty ?

Reply:
Yes, the additional else case changes the behavior to show the translatable message that maps to each AppInstallStatus when the error message is otherwise empty. Removing this case would restore the prior behavior, i.e. not raising a notification unless there's an error message. I can put it back that way if you prefer.

Reply:
Yeah my guess here is we only want to show notifications when there is something more around errors than what's presented on screen to the user. So showing notifications in case of no error messages might get redundant in usage or result in too many error notifications. So probably best to preserve the original behaviour unless we do some testing around the change to see if it causes redunandant/many notifications.